claim-check/code-samples/sample-2/client-consumer/EventReader.cs

```diff
@@ -0,0 +1,134 @@
+                    return;
+                }
+            }
+
+            string storageConnectionString = ConfigurationManager.AppSettings["StorageConnectionString"];
+            string eventhubConnectionString = ConfigurationManager.AppSettings["EventHubConnectionString"];
+            downloadDestination = ConfigurationManager.AppSettings["DownloadDestination"];
+            string blobContainerName = ConfigurationManager.AppSettings["BlobContainerName"];
+            Console.WriteLine("Connecting to Storage account...");
+            BlobContainerClient blobContainerClient = new BlobContainerClient(storageConnectionString, blobContainerName);
+            Console.WriteLine("Connecting to EventHub...");
+            processor = new EventProcessorClient(blobContainerClient, EventHubConsumerClient.DefaultConsumerGroupName, eventhubConnectionString);
+        }
+
+        //As an example, we'll just log the exception to the console in try/catch block.
+        //For real-world scenarios, you should take action appropriate to your application.
+        public async Task ProcessMessages(CancellationToken cancellationToken)
+        {
+            Console.WriteLine("The application will now start to listen for incoming messages.");
+
+            Task partitionInitializingHandler(PartitionInitializingEventArgs eventArgs)
+            {
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return Task.CompletedTask;
+                }
+                try
+                {
+                    var utcNow = DateTime.UtcNow;
+                    eventArgs.DefaultStartingPosition = EventPosition.FromEnqueuedTime(utcNow);
+                    Console.WriteLine($"Initialized partition: { eventArgs.PartitionId }");
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine($"An error was observed while initializing partition: { eventArgs.PartitionId }.  Message: { ex.Message }");
+                    Console.WriteLine();
+                }
+                return Task.CompletedTask;
+            }
+
+            async Task processEventHandlerAsync(ProcessEventArgs eventArgs)
+            {
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+                try
+                {                  
+                    string body = Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray());
+                    var jsonMessage = JArray.Parse(body).First;
+                    Uri uploadedUri = new Uri(jsonMessage["data"]["url"].ToString());
+                    Console.WriteLine("Blob available at: {0}", uploadedUri);
+                    BlockBlobClient blockBlob = new BlockBlobClient(uploadedUri);
+                    string uploadedFile = Path.GetFileName(jsonMessage["data"]["url"].ToString());
```

Comment:
nit:  Why not use `uploadedUri` here, since you loaded it with the same JSON value on line 83.